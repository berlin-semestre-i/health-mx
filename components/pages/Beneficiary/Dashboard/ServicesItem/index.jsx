import React from 'react'
import { Grid, Image } from 'semantic-ui-react'
import styled from 'styled-components'
import Link from 'next/link'
import media from 'styled-media-query'

const ServiceItem = ({ service }) => (
  <ServiceRow>
    <IconContainer mobile={16} tablet={3} computer={4}>
      <Link as="" href={service.link}><a target="_blank"><Icon src={service.icon}/></a></Link>
    </IconContainer>
    <ServiceInfoContainer mobile={16} tablet={13} computer={12}>
      <Link as="" href={service.link}><a target="_blank">{ service.name }</a></Link>
    </ServiceInfoContainer>
  </ServiceRow>
)

export default ServiceItem

const ServiceRow = styled(Grid)`
  >.column:not(.row):last-child {
    padding-top: 1.5rem;
    margin-bottom: 1.5rem;
  }
`

const IconContainer = styled(Grid.Column)`
  text-align: center;

  ${media.between('medium','1025px')`
    [class*="two wide computer"]&&&.column {
      width: 18.75%!important;
    }
  `}

  ${media.lessThan('medium')`
    &&&.column:not(.row) {
      padding-bottom: 0;
    }
  `}
`
const ServiceInfoContainer = styled(Grid.Column)`
  ${media.between('medium','1025px')`
    [class*="fourteen wide computer"]&&&.column {
      width: 81.25%!important;
    }
  `}
`
const Icon = styled(Image)`
  && {
    border-radius: 30px;
    max-width: 55px;
    max-height: 55px;
  }
`