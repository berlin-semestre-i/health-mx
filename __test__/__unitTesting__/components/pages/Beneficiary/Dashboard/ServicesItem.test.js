import React from 'react'
import { shallow, mount } from 'enzyme'
// eslint-disable-next-line max-len
import ServicesItem from '../../../../../../components/pages/Beneficiary/Dashboard/ServicesItem/index'

const testService = {
  name: 'NÃºmero de Seguro Social',
  icon: '../../../../../../static/images/NSSIcon.png',
  link: 'http://bit.ly/imssNSS',
}

describe('Beneficiary Services', () => {
  it('renders', () => {
    const component = shallow(<ServicesItem service={testService} />)
    expect(component).toMatchSnapshot()
  })
})

it('has an icon', () => {
  const wrapper = mount(<ServicesItem service={testService} />)
  expect(wrapper.find(`img[src="${testService.icon}"]`).length).toBe(1)
  wrapper.unmount()
})

it('has a link', () => {
  const wrapper = mount(<ServicesItem service={testService} />)
  expect(wrapper.find(`a[href="${testService.link}"]`).length).toBe(2)
  wrapper.unmount()
})