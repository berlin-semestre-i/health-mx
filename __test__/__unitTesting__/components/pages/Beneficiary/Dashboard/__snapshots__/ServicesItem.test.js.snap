// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beneficiary Services renders 1`] = `
<ServicesItem__ServiceRow>
  <ServicesItem__IconContainer
    computer={4}
    mobile={16}
    tablet={3}
  >
    <Link
      as=""
      href="http://bit.ly/imssNSS"
    >
      <a
        target="_blank"
      >
        <ServicesItem__Icon
          src="../../../../../../static/images/NSSIcon.png"
        />
      </a>
    </Link>
  </ServicesItem__IconContainer>
  <ServicesItem__ServiceInfoContainer
    computer={12}
    mobile={16}
    tablet={13}
  >
    <Link
      as=""
      href="http://bit.ly/imssNSS"
    >
      <a
        target="_blank"
      >
        NÃºmero de Seguro Social
      </a>
    </Link>
  </ServicesItem__ServiceInfoContainer>
</ServicesItem__ServiceRow>
`;
